crypto/rand/randfile.c

```diff
@@ -162,9 +168,16 @@ int RAND_load_file(const char *file, long bytes)
             n = (bytes < BUFSIZE) ? (int)bytes : BUFSIZE;
         else
             n = BUFSIZE;
+
         i = fread(buf, 1, n, in);
-        if (i <= 0)
+        if (i <= 0) {
```

Comment:
Formally speaking this is, and was, incorrect. `fread` returns `size_t`, i.e. it can't be negative and condition should be `i == 0`. Well, manual even says that `i` being less than `n` can be sign of error condition too... In other words some extra consideration might be required...